python/pyspark/sql/session.py

```diff
@@ -384,17 +384,15 @@ def _createFromLocal(self, data, schema):
 
         if schema is None or isinstance(schema, (list, tuple)):
             struct = self._inferSchemaFromList(data)
+            converter = _create_converter(struct)
```

Comment:
This [`_create_converter` method](https://github.com/davies/spark/blob/c0ad8668ba22e51b07ba08b8e19c312783cd1b87/python/pyspark/sql/types.py#L1054) is confusingly-named: what it's actually doing here is converting `data` from a dict to a tuple in case the schema is a StructType and data is a Python dictionary.